core/src/main/kotlin/org/evomaster/core/output/TestCaseWriter.kt

```diff
@@ -728,29 +785,25 @@ class TestCaseWriter {
         }
     }
 
-    /**
-     * [applyEscapes] currently sets up the string for printing.
-     * This includes escaping special chars for java and kotlin.
-     * Currently, Strings containing "@" are split, on the assumption (somewhat premature, admittedly) that
-     * the symbol signifies an object reference (which would likely cause the assertion to fail).
-     * TODO: Tests are needed to make sure this does not break.
-     */
-
-    /*
-
-
-    private fun applyEscapes(string: String): String {
-        val timeRegEx = "[0-2]?[0-9]:[0-5][0-9]".toRegex()
-        val ret = string.split("@")[0] //first split off any reference that might differ between runs
-                .split(timeRegEx)[0] //split off anything after specific timestamps that might differ
-                .replace("""\\""", """\\\\""")
-                .replace("\"", "\\\"")
-                .replace("\n", "\\n")
-                .replace("\r", "\\r")
-
+    fun flattenForAssert(k: MutableList<*>, v: Any): Map<MutableList<*>, Any>{
```

Comment:
does this method need to be public?
if so, provide some docs for it, eg explaining its inputs and outputs

Reply:
I don't think it will be used elsewhere, so I made it private for now. 
I also added a little bit of an explanation as to what it does. It's more of an auxiliary method (I looked for ready-made stuff that did similar things, but ultimately it was less time-consuming just to write the thing).